test/basic/basic.go

```diff
@@ -93,6 +93,22 @@ func (s *TestSuite) TestBasics(t *testing.T) {
 	resp, err = subClient.Recv() // Now get the addition notification
 	assert.NoError(t, err)
 	assert.Equal(t, "201", resp.GetUpdate().Update[0].Path.Elem[1].Key["port"])
+
+	wg2 := sync.WaitGroup{}
+	wg2.Add(5) // setting counter to 5 (until at least 5 hosts are found..)
+
+	for i, device := range dresp.Devices {
```

Comment:
I would connect only to the leaves here since spines will have no hosts. So just add an 'if' inside the for loop and test for prefix 'leaf' on the device name.

Reply:
Addressed